File: brixen-java/src/test/java/org/brixen/test/decorator/bean/LoadableBeanProviderTest.java
package org.brixen.test.decorator.bean;

import org.brixen.bean.LoadableBean;
import org.brixen.decorator.bean.LoadableBeanProvider;
import org.testng.annotations.Test;

@Test(groups = {"ProviderTestGroup"})
public class LoadableBeanProviderTest {

    @SuppressWarnings("ConstantConditions")
    @Test(groups = {"ExpectedExceptionsTestGroup", "ProviderExpectedExceptionsTestGroup",
            "LoadableBeanProviderExpectedExceptionsTestGroup"},
            expectedExceptions = { NullPointerException.class },
            expectedExceptionsMessageRegExp = "Cannot construct a LoadableBeanProvider with a null bean"
    )
    public void shouldThrowExceptionForNullBean() {
        new LoadableBeanProvider<>((LoadableBean)null);
    }
}

File: brixen-dotnet/test/decorator/bean/Org.Brixen.Decorator.Bean.Tests/LoadableBeanProviderTest.cs
using NUnit.Framework;
using System;
using Org.Brixen.Bean;
using Org.Brixen.Decorator.Bean;

namespace Org.Brixen.Decorator.Bean.Tests {
	
	[TestFixture]
	[Category("LoadableBeanProviderTestGroup")]
	public class LoadableBeanProviderTest {
		
		[Test]
		[Category("ExpectedExceptionsTestGroup")]
		[Category("ProviderExpectedExceptionsTestGroup")]
		[Category("LoadableBeanProviderExpectedExceptionsTestGroup")]
		[ExpectedException("System.ArgumentNullException", ExpectedMessage = "Cannot construct a " + 
			"LoadableBeanProvider with a null bean", MatchType=MessageMatch.Regex)]
		public void shouldThrowExceptionForNullBeanElement() {
			new LoadableBeanProvider<ILoadableBean>(null);
		}
	}	
}

